pkg/linear-pools/contracts/midas-linear-pool/MidasLinearPoolRebalancer.sol

```diff
@@ -18,62 +18,48 @@ pragma experimental ABIEncoderV2;
 import "./interfaces/ICToken.sol";
 
 import "@balancer-labs/v2-interfaces/contracts/pool-utils/ILastCreatedPoolFactory.sol";
+//import "@balancer-labs/v2-interfaces/contracts/solidity-utils/openzeppelin/IERC20.sol";
 
 import "@balancer-labs/v2-solidity-utils/contracts/math/FixedPoint.sol";
 import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/ERC20.sol";
 import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/SafeERC20.sol";
 
-import "./contracts/LinearPoolRebalancer.sol";
+import "@balancer-labs/v2-pool-linear/contracts/LinearPoolRebalancer.sol";
 
 contract MidasLinearPoolRebalancer is LinearPoolRebalancer {
-    using FixedPoint for uint256;
     using SafeERC20 for IERC20;
+    using FixedPoint for uint256;
 
     uint256 private immutable _divisor;
 
     // These Rebalancers can only be deployed from a factory to work around a circular dependency: the Pool must know
     // the address of the Rebalancer in order to register it, and the Rebalancer must know the address of the Pool
     // during construction.
-    constructor(IVault vault, IBalancerQueries queries)
-        LinearPoolRebalancer(ILinearPool(ILastCreatedPoolFactory(msg.sender).getLastCreatedPool()), vault, queries)
-    {
-        ILinearPool pool = ILinearPool(ILastCreatedPoolFactory(msg.sender).getLastCreatedPool());
-        ERC20 mainToken = ERC20(address(pool.getMainToken()));
-        ERC20 wrappedToken = ERC20(address(pool.getWrappedToken()));
-
-        // The CToken function exchangeRateCurrent returns the rate scaled to 18 decimals.
+    constructor(
+        IVault vault,
+        IBalancerQueries queries,
+        address mainToken,
+        address wrappedToken
+    ) LinearPoolRebalancer(ILinearPool(ILastCreatedPoolFactory(msg.sender).getLastCreatedPool()), vault, queries) {
+        // The CToken function exchangeRateHypothetical returns the rate scaled to 18 decimals.
         // when calculating _getRequiredTokensToWrap, we receive wrappedAmount in the decimals
         // of the wrapped token. To get back to main token decimals, we divide by:
         // 10^(18 + wrappedTokenDecimals - mainTokenDecimals)
-        _divisor = 10**(18 + wrappedToken.decimals() - mainToken.decimals());
+        _divisor = 10**(18 + ERC20(wrappedToken).decimals() - ERC20(mainToken).decimals());
     }
 
     function _wrapTokens(uint256 amount) internal override {
-        // Depositing from underlying (i.e. DAI, USDC, etc. instead of cDAI or cUSDC). Before we can
-        // deposit however, we need to approve the wrapper (cToken) in the underlying token.
         _mainToken.safeApprove(address(_wrappedToken), amount);
         ICToken(address(_wrappedToken)).mint(amount);
```

Comment:
We should check the output like we do in the [`CompoundV2Wrapping` relayer](https://github.com/balancer/balancer-v2-monorepo/blob/b3b919775903450c5c3b4d22f1daf14689ce9cb7/pkg/standalone-utils/contracts/relayer/CompoundV2Wrapping.sol#L42).